### ISSUE TITLE
1.10-rc3 PEAR Mail::mimeDecode via API HTTP ticket.email causes Empty Thread

### ISSUE THREAD

**Issue by ooglek** [osTicket/osTicket#3278]
Using PHP 7.0.8, OSTicket 1.10-RC3. Piping emails from server to OSTicket through the HTTP API, using /api/tickets.email

I noticed that the script I was piping email data to was showing new threads, but the threads were empty. I found the code that set $seen, which is why further attempts to overwrite the thread with failed. I saw an error in my error_log:

> [19-Jul-2016 04:03:04 UTC] PHP Warning:  preg_replace(): The /e modifier is no longer supported, use preg_replace_callback instead in /usr/local/www/vhosts/osticket/include/pear/Mail/mimeDecode.php on line 648

So I rewrote it with preg_replace_callback(), manually deleted the thread from ost_thread_entry and ost_thread_entry_email, then piped the email back through to OST, and the email content showed up.

So this is a bug in software OSTicket depends on but distributes. Maybe there is an upstream update to Mail::MimeDecode? I'll let you decide. This fixes the issue for 1.10-RC3.

```diff
--- /tmp/mimeDecode.php 2016-07-19 04:59:44.739132000 +0000
+++ /tmp2/mimeDecode.php    2016-07-19 04:59:16.334200000 +0000
@@ -645,7 +645,7 @@
         $input = preg_replace("/=\r?\n/", '', $input);

         // Replace encoded characters
-               $input = preg_replace('/=([a-f0-9]{2})/ie', "chr(hexdec('\\1'))", $input);
+        $input = preg_replace_callback('/=([a-f0-9]{2})/i', function($m) { return chr(hexdec($m[0])); }, $input);

         return $input;
     }
```

Since mimeDecode hasn't been touched since 2010, maybe time to replace it? 

https://packagist.org/packages/php-mime-mail-parser/php-mime-mail-parser Seems to support PHP 7.

I'd submit a pull request but since this isn't core OSTicket code, I'm not sure the team wants to take on maintaining a branch of no-longer-maintained 3rd party PEAR code.

**Comment by ooglek** [comment]
The issue may also have been that there were special characters in the body of the email, which is why some emails got through fine, and others got a blank email. The email in question **was** MIME-encoded with multiple parts.

**Comment by mfelber** [comment]
4-byte characters truncate emails: https://github.com/osTicket/osTicket/issues/1475

**Comment by ooglek** [comment]
Emoji support does not fix this issue. #3244 fixed utf8mb4 support for the database. 

The email that caused this issue was in 2 Mime Parts.

X-OriginatorOrg: hotmail.com
MIME-Version: 1.0
Content-Type: multipart/alternative;
    boundary="_000_CO1PR16MB098225D1BF37AA6EBBBD933BC5360CO1PR16MB0982namp_"

The first was plain text, the second was HTML. The failure was on the mimeDecode.php line. The thread inserted into the database, but the content of the inserted thread was empty (I think there was a single dash or something).  

#1475 is unrelated due to the fact that was an encoding issue and the DB failed to insert. My issue is that the message inserted but the content/body was empty. Two very different bugs.

And I found the bug, fixed it, tested it with the failed message, had it succeed, then provided the diff for a fix.

**Comment by ooglek** [comment]
I believe the issue is when there are MIME-encoded characters such as "=20" (a space) in the email, whether it is the text portion or the HTML portion. This causes the email to be inserted into the DB as an empty response. The one line change above fixes the issue.

**Comment by martin-rueegg** [comment]
Mail::mimeDecode has been updated to [v1.5.6](http://pear.php.net/package/Mail_mimeDecode/download/) and is supposed to address the issue (see their change log and bug [#20027](http://pear.php.net/bugs/bug.php?id=20027).

However, simply upgrading to the [new version](https://github.com/osTicket/osTicket/pull/3362), does not solve the issue (see #3361).

The above fix by @ooglek does work.

Maybe v1.5.6 has introduced a new issue elsewhere? I even tried to apply @ooglek's fix to the v1.5.6 version, but still no luck ...

**Comment by phpHavok** [comment]
@ooglek's solution worked for me. Clean and simple fix.

**Comment by dahlenu** [comment]
I also had this problem (osTicket v1.10.1). @ooglek's solution fixed it.

**Comment by lluisYasT** [comment]
I also had the same problem with osTicket v1.10.4. I applied @ooglek 's solution and it is working as expected now.

**Comment by helpwise** [comment]
Fixed my issue too.  Thanks :)

**Comment by adhiemsumitro** [comment]
This is what i've done :
1.  Backup the mimeDecode.php
`mv /var/www/html/ticket/include/pear/Mail/mimeDecode.php /var/www/html/ticket/include/pear/Mail/mimeDecode.php.bak`
2.  Update the php PEAR
`sudo pear channel-update pear.php.net`
3. Upgrade PEAR mailDecode 
`pear install Mail_mimeDecode`
4. Maybe thi isi needed ? 
`sudo apt install php-mail-mime php-mime-type`
5 symlink new mimeDecode.php
`ln -s /usr/share/php/Mail/mimeDecode.php /var/www/html/ticket/include/pear/Mail/mimeDecode.php`

Tested creating new-ticket from mail and the mail alert `~{message}` shown as expected

**Comment by LeCoyote** [comment]
Came here to confirm the bug and its resolution. Running v1.11 on PHP 7.3 here, and replacing osTicket's mimeDecode with PEAR's version 1.5.6 solved the issue. So basically, #3362 from over 2 years ago never got merged into a release, which is why the problem still exists today.

For completeness' sake, in our case, the mail client causing the problem was Apple iPhone's.

**Comment by JediKev** [comment]
@LeCoyote

v1.11 doesn’t support PHP 7.3, only v1.12 does. v1.11 has been dropped due to security issues so please upgrade to v1.12 ASAP. Also in v1.12 we addressed this issue.

Cheers.

**Comment by LeCoyote** [comment]
A-ha. Didn't know that. Will do, thanks for the heads-up!

**Comment by libussa** [comment]
I still get the issue when receiving emails from hotmail with accented characters (php 7.2, ost 1.12 running in docker)

**Comment by JediKev** [comment]
@libussa 

Have you tried this:
https://github.com/osTicket/osTicket/pull/4932

Cheers.

**Comment by libussa** [comment]
> @libussa
> 
> Have you tried this:
> #4932
> 
> Cheers.

I just did, it didn't solve the issue. 
This is an example of a message that produces a ticket with an empty body:

```
Subject: test ticket 5, ignore
Thread-Topic: test ticket 5, ignore
Thread-Index: AQHVPowAwj6cT6yEq0q7wCRpdwnoKw==
Date: Fri, 19 Jul 2019 23:45:18 +0000
Message-ID: <VI1PR04MB51179C515A0B359E330ED45FA8CB0@VI1PR04MB5117.eurprd04.prod.outlook.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-incomingtopheadermarker: OriginalChecksum:4B79F1BD820C0DF28FD2175A8ED92B54787ADCE78FBE3E40ABB5B80EC6FEC2E7;UpperCasedChecksum:FD244BA5747E52B5442B9B2F1A4A885746EB77E307D05605D0123189F54AB315;SizeAsReceived:6497;Count:40
x-tmn: [AFJUgytn7C3f08FcD/xOKhtOm8ig1vUftxFffNHA81U=]
x-ms-publictraffictype: Email
x-incomingheadercount: 40
x-eopattributedmessage: 0
x-microsoft-antispam: BCL:0;PCL:0;RULEID:(2390118)(5050001)(7020095)(20181119110)(201702061078)(5061506573)(5061507331)(1603103135)(2017031320274)(9118020)(2017031322404)(2017031323274)(2017031324274)(1601125500)(1603101475)(1701031045);SRVR:HE1EUR02HT069;
x-ms-traffictypediagnostic: HE1EUR02HT069:
x-microsoft-antispam-message-info: J29BMawgqsmOEqWsTuPcqgxqsZgd9yflEzDcFx81APDMvZhdCKLXtlWsU/pOIRn6fNTjGDWoIMgxOkK3q2PbPyN122MCdhzmpawcVJ39cfPw1Tg9mbQaJw0tqnTu87RkktWfo6YgzgfpH6l9O8ZVU1MA8Ok2pwZCGlve64dpr8Dxf0KSRD4Goxj68mBCoNe+
Content-Type: multipart/alternative; boundary="_000_VI1PR04MB51179C515A0B359E330ED45FA8CB0VI1PR04MB5117eurp_"
MIME-Version: 1.0
X-OriginatorOrg: hotmail.com
X-MS-Exchange-CrossTenant-RMS-PersistedConsumerOrg: 00000000-0000-0000-0000-000000000000
X-MS-Exchange-CrossTenant-Network-Message-Id: af53e208-382d-4d41-c5d0-08d70ca3283c
X-MS-Exchange-CrossTenant-rms-persistedconsumerorg: 00000000-0000-0000-0000-000000000000
X-MS-Exchange-CrossTenant-originalarrivaltime: 19 Jul 2019 23:45:18.9289 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Internet
X-MS-Exchange-CrossTenant-id: 84df9e7f-e9f6-40af-b435-aaaaaaaaaaaa
X-MS-Exchange-Transport-CrossTenantHeadersStamped: HE1EUR02HT069

--_000_VI1PR04MB51179C515A0B359E330ED45FA8CB0VI1PR04MB5117eurp_
Content-Type: text/plain; charset="iso-8859-1"
Content-Transfer-Encoding: quoted-printable

text with =E8=E9=EA accents
other accents =E0=E2=E6=E4=E9

--_000_VI1PR04MB51179C515A0B359E330ED45FA8CB0VI1PR04MB5117eurp_
Content-Type: text/html; charset="iso-8859-1"
Content-Transfer-Encoding: quoted-printable

<html>
<head>
<meta http-equiv=3D"Content-Type" content=3D"text/html; charset=3Diso-8859-=
1">
<style type=3D"text/css" style=3D"display:none;"> P {margin-top:0;margin-bo=
ttom:0;} </style>
</head>
<body dir=3D"ltr">
<div style=3D"font-family: Calibri, Arial, Helvetica, sans-serif; font-size=
: 12pt; color: rgb(0, 0, 0);">
text with =E8=E9=EA accents</div>
<div style=3D"font-family: Calibri, Arial, Helvetica, sans-serif; font-size=
: 12pt; color: rgb(0, 0, 0);">
other accents =E0=E2=E6=E4=E9<br>
</div>
</body>
</html>

--_000_VI1PR04MB51179C515A0B359E330ED45FA8CB0VI1PR04MB5117eurp_--
```

This is happening from @outlook.com, @hotmail.com, @live.com addresses 
The text sent is:
```
text with èéê accents
other accents àâæäé
````
it works fine from a gmail address

**Comment by JediKev** [comment]
@libussa

You didn’t apply the changes correctly then because that’s your issue. It’s the charset of `iso-8859-1` that makes them empty.

Make sure you apply the changes fully/correctly, restart Apache/NGINX, clear the browser cache, and retest. Also this doesn’t apply to old tickets only new tickets after the changes are applied.

Cheers.

**Comment by libussa** [comment]
@JediKev 
I applied the diff, checked that the file was correct, and started nginx and php-fpm. 
I still get the (empty) body on a newly created ticket :(

**Comment by JediKev** [comment]
@libussa

Ooo nice catch. I will add this case to the regex when I get the chance.

Cheers.

**Comment by jortkoopmans** [comment]
@JediKev ; What do you mean with 'nice catch' and referencing the regex? I think you're misunderstanding what @libussa is saying (or I'm confused myself :wink:);

- The case handled here is about ISO-8859-1 (not trailing i), I've checked the regex and it should match already?
- All options provided in this thread have been tried to no avail. I've verified it's not related to multipart (single part 8859-1 also fails).

I'm not really sure whether the problem is the same as the OP, given 1.12.x already includes the various fixes...
We might be looking at another edge case here? I'm experiencing this with OS 1.12.3, nginx 1.16.1 ,PHP7.3.11, using docker...

**Comment by JediKev** [comment]
@jortkoopmans

I took a quick glance without recognizing we already have it.

**Comment by kswtch** [comment]
> I'm not really sure whether the problem is the same as the OP, given 1.12.x already includes the various fixes...
> We might be looking at another edge case here? I'm experiencing this with OS 1.12.3, nginx 1.16.1 ,PHP7.3.11, using docker...

Probably, I have the same issue as described by @jortkoopmans with osTicket 1.14.1, nginx 1.16.1, PHP 7.3.12 using docker image campbellsoftwaresolutions/osticket

**Comment by JediKev** [comment]
@kswtch 

You'll have to post a full raw example email (including headers) that's failing in order for us to attempt to replicate the issue as we cannot replicate the same.

Cheers.

**Comment by jortkoopmans** [comment]
See mailheaders attached. I redacted the domainname only. Note that I capture this mail by having a mailforward on the mailbox, because osticket will read the email and purge it from the 'info' emailbox. But the headers and 8859-1 are identical (meaning it ends up with '(empty)' as its content).
@libussa already posted similar full mail that won't be parsed properly....What I'm trying to get across is that this is happening with the correct patches already being applied.

I'm suspecting its related to configuration specifics on either nginx or php-fpm.

[20191118_mailheaders.txt](https://github.com/osTicket/osTicket/files/4020497/20191118_mailheaders.txt)

**Comment by JediKev** [comment]
@jortkoopmans 

I just tested your example by piping and it came in just fine for me:
<img width="590" alt="Screen Shot 2020-01-03 at 13 06 31" src="https://user-images.githubusercontent.com/11823401/71743336-eba1ae80-2e29-11ea-8aa5-227691e61388.png">

Can you try piping your example the test to see if it works for you via Pipe? From command line run `php /path/to/osticket/api/pipe.php < /path/to/20191118_mailheaders.txt` to pipe the email.

Cheers.

**Comment by jortkoopmans** [comment]
Thank you for your response @JediKev , much appreciated.
TLDR; The suggested cli and pipe gives the same result with a transaction showing "(empty)".

I've wrestled with this a bit, but this is what I've observed when following your suggestion;

- When processing a message that already exists in the ticket, I don't observe anything happening. The php command (as described above) completes without any messages/stdout.
- When piping a new message with the same formatting (in my test Content-Type: text/plain; charset="iso-8859-1"), the php command completes without any messages/stdout, but a new transaction is recorded in the ticket, with the "(empty)" content. This is identical to mail received directly via the imap hook as described earlier.

What I therefore think is happening is that an existing transaction is filtered on mail ID or something, therefore not duplicating it in the ticket. A new message processes OK, unfortunately also without any errors/logging. I've looked at the various log files, but nothing seems to be logged there either.

![screenshot_20200107](https://user-images.githubusercontent.com/1008482/71933382-4593d280-31a2-11ea-9b44-31040f761702.png)


php --version
PHP 7.3.11 (cli) (built: Oct 25 2019 03:29:07) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.11, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.3.11, Copyright (c) 1999-2018, by Zend Technologies

What should I be looking at to further investigate this?

**Comment by JediKev** [comment]
@jortkoopmans 

This is very strange to me as I have the same setup (with v1.14.1) and this works every time for me.
<img width="373" alt="Screen Shot 2020-01-07 at 16 11 12" src="https://user-images.githubusercontent.com/11823401/71933677-5b84a180-3168-11ea-8bfe-bba0adfe6234.png">
<img width="550" alt="Screen Shot 2020-01-07 at 16 08 05" src="https://user-images.githubusercontent.com/11823401/71933528-f92ba100-3167-11ea-8f39-3aa4c1c90989.png">

**Questions:**
-  Are you using v1.12 or v1.14?
- Are you piping an email that is new and not related to any tickets in your helpdesk (like a new email) or are you piping a response to an existing ticket? If the latter, please test with a completely new email with accents to see if this works. Maybe it's only messages to existing tickets messing up?

**UPDATE:**
I tested this with my same setup above but with v1.12.5 and this worked for me again...I think the issue is you are using an older version in the v1.12 series that does not contain the patch for this. Please upgrade to v1.12.5 and see if this addresses your issue.

Cheers.

**Comment by jortkoopmans** [comment]
Thanks for the feedback and ideas.
Unfortunately I can't confirm your hypothesis.
I've tried to;

- Upgrade to v1.14.1 (upgrade process went flawlessly!)
- Piping, replying via mail, creating new tickets, but this text formatting is never showing up in tickets. Note that all other mails work fine (also in tickets with empty response transactions), so I don't think tickets are being corrupted.

![OSTicket_20200110](https://user-images.githubusercontent.com/1008482/72187323-a8ca7280-33f7-11ea-9ffc-4716cca6910c.png)
There's still some differences in versions with your instance (nginx, php minor, mariadb vs mysql?)...but I simply don't know where to start without any errors/logging.
I know the docker image is not supported, but if you would be able to reproduce this problem there, that would be a breakthrough...

**Comment by JediKev** [comment]
@jortkoopmans 

Question, did you copy/replace the new files on top of the old files? Or did you make the site folder empty, upload the new files, and copy over the config file? I’m wondering if you copy/replaced the files because I’ve seen plenty of times before where it doesn’t copy over some of the updated files which causes issues as some parts of the code are updated but other parts are not (that are supposed to be).

Cheers.

**Comment by jortkoopmans** [comment]
I'm convinced that the core osticket files are new, as I'm using the docker setup (which aims to guarantee a consistent file set).
State is retained in:

- Database (/var/lib/mysql)
- Plugins (osticket/include/plugins...but the container only installs ldap auth)
- Logs (/var/log/nginx)
- Configuration via environment variables (db info, INSTALL_URL etc)

**Comment by AirPett** [comment]
@libussa, @jortkoopmans and @kswtch I'm having the exact same setup and issue as you. I tried both the campbellsoftwaresolutions/osticket and the devinsolutions/osticket. Both of them sometimes create the tickets successfully from my postfix IMAP mailbox, but most of the time I end up with (empty).

Here is my version information:
![image](https://user-images.githubusercontent.com/32071163/74025030-be599a80-49ab-11ea-848c-298eb6caef9b.png)

Did you ever manage to solve this using the prebuilt Docker containers? I'd love not having to try build a new container from scratch. I'll though still try to run apache and php on bare metal instead of a container and manually install osticket there to see if it works.

**Comment by jortkoopmans** [comment]
@AirPett ; Unfortunately I still experience this problem and honestly I don't have a clear idea on where to begin to find the cause for it. I guess it's somewhat falling through the cracks, because most likely it isn't strictly an OsTicket issue. Instead it's probably having something to do with the packages and versions used by the Docker image (which is not supported by Enhancesoft).

**Comment by AirPett** [comment]
@jortkoopmans Ok I downloaded osTicket v1.14.1 from the official download page with a couple of languages and plugins.

I spun up a [mariadb:10](https://hub.docker.com/_/mariadb) and configured the database with environment variables. I also started my own [gitlab.yobitti.fi:5001/yobitti/apache-php](https://gitlab.yobitti.fi/yobitti/apache-php) and mounted the ost/upload/ to /var/www/html.

After that I just quick and dirty started a bash shell in the container and installed the required PHP extensions (mysqli, gd, intl, imap, opcache, etc.) using the installer mentioned in my apache-php's README and Dockerfile.

I then just browsed to the new osTicket and went through the installer by manually entering DB and other info, went straight to Admin->Emails->Settings and enabled IMAP fetch and autocron as well as entered the IMAP server details and credentials to the Emails->support@example.com.

AND: HOORAY! After sending an email from the earlier broken clients the ticket was successfully created and even HTML content was displayed without problems (see images below)! (broken clients = creating empty tickets when mail sent from them; For me it didn't happen with all email senders i.e. mail sent from Android Gmail app never turned out empty, but Microsoft Office Outlook O365 newest version and Outlook Web App (OWA) always resulted in empty emails).

### Here is the new version table:
![image](https://user-images.githubusercontent.com/32071163/74075813-2eeccf80-4a1d-11ea-9ed1-492e12d786d9.png)

I used the same email inbox for both, the old and new installation, and set the old to fetch with 5 minute cron and the new with 1 minute while not doing anything for the emails --> I was able to compare the two systems with the exact same email messages.

### The campbellsoftwaresolutions/osticket:
![image](https://user-images.githubusercontent.com/32071163/74075908-bcc8ba80-4a1d-11ea-9ca1-3b14c13b49c9.png)

### My apache-php container and half manual osTicket installation (the exact same source email message):
![image](https://user-images.githubusercontent.com/32071163/74075984-15985300-4a1e-11ea-9975-ff6571b24909.png)

There was a working mailto: -link under the green paint in the second image.

I know that the installation method is far from ideal for now, but I'll build and publish a docker container that has everything inside (osTicket, php extensions, bootstrap script to configure osTicket with environment variables) as soon as I have time, hopefully this weekend or next week.

**Comment by AirPett** [comment]
One more note to add to the previous comment: This also solved 2 other big problems for me:
- The campbell container didn't fully support nordic characters (öäå in email created tickets' subjects showed as ? in addition to the message being (empty))
- I was unable to upload and attach any files larger than a couple hundred KBs no matter what settings or php.ini files I modified to set the limit (upload and post) to 100MB. Now I was able to upload a 1,95MB file having the PHP limit at 2MB.

**Comment by marchermans** [comment]
@AirPett Did you ever publish the container? Cause i ran into the same issue and it would be amazing if you had it somewhere.

**Comment by AirPett** [comment]
@OrionDevelopment It's still a bit of work in progress as due to other work I haven't had time to continue much further from where I left. I'll give an update as soon as I make some progress.

**Comment by marchermans** [comment]
Okey thanks for the heads up.

**Comment by marchermans** [comment]
@AirPett I got it to run. If you want i can publish my stuff.

**Comment by AirPett** [comment]
@OrionDevelopment That would be very nice of you if you could publish it.

**Comment by mausquirk** [comment]
@OrionDevelopment @AirPett 
Any news about our new container? 
Is there somewhere a repository to contribute?

**Comment by marchermans** [comment]
Ahh sorry, i forgot to make the repo public. Will do as soon as possible. And post a link here. Thanks for the reminder.

**Comment by marchermans** [comment]
Here we go: @mausquirk : https://gitlab.ldtteam.com/marc.hermans/osticket-docker

This git repository contains all the Dockerfiles and even a docker-compose to get going quickly.
By default it uses the 1.14.1 branch of this repository. But by specifying the git repo and branch as version on the builder image you can specify your own repo and version to use.

Hope this helps. Orion.

**Comment by jortkoopmans** [comment]
@OrionDevelopment ; This does not resolve for me. Are you referencing some company internal resource?

**Comment by marchermans** [comment]
No this is a public website.
However it does not have the greatest of uptimes. It gets regularly daily updates. And when security updates arise the host gets those nearly immediatly. So sometimes you might hit a moment where it is down.

I just checked it should be up.

**Comment by jortkoopmans** [comment]
Ah yes it works now, thanks.

**Comment by kosli** [comment]
@OrionDevelopment  your link is not working (404 page not found). Can you share it somehwere else? I would like to take a look and see if I find a difference to the not working two docker containers.

**Comment by marchermans** [comment]
I know
We are moving servers....
And to say the least it is not going as well as we hoped

**Comment by peptoniet** [comment]
@OrionDevelopment Hello my friend.  Did you manage to share your dockerfiles, etc...?  Thanks a lot!

**Comment by marchermans** [comment]
@peptoniet Sorry for the delay.
They are online now: https://gitlab.ldtteam.com/marc.hermans/osticket-docker

**Comment by yurividal** [comment]
@OrionDevelopment do you still have that docker image? Your gitlab resolves fine, but the actual docker image is hosted on "docker.ldtteam.com" and this seems to be down. 
Could you publish it to dockerhub?

**Comment by marchermans** [comment]
Ah yeah we migrated to a new docker host.
Give me a second to publish that for you

**Comment by marchermans** [comment]
@yurividal Updated the images and the repository.

They are now hosted on container.ldtteam.com, since we updated to a new cluster and had to move the repository.
Sorry for the inconvenience.

Greets,

Orion

**Comment by yurividal** [comment]
@OrionDevelopment thanks dude. I'll have a look. I have been running the campbellsoftwaresolutions container for over a year, but started to have problems with tickets getting (empty) body. I'll try the apache version to see if it solves.

**Comment by yurividal** [comment]
@OrionDevelopment how do you handle the cron for email fetch on your container?

**Comment by marchermans** [comment]
@yurividal We use the cron api to invoke it from either a server (which we did for the longest time), but have now switched to a simple bash container which invokes the api once every second or so.

**Comment by yurividal** [comment]
Cool, thanks. I'm not very proficient with docker, but I am being able to modify your files and have the cron run on the main container. Now I will add an automatic configuration file, which reads from env variables.

**Comment by marchermans** [comment]
@yurividal it would be nice if you made a pr for it? Or send me a changelist

**Comment by yurividal** [comment]
> @yurividal it would be nice if you made a pr for it? Or send me a changelist
 Will do. I already have the cron part working. It's on my github.
Now I'm trying to figure out how to get the runtime environment variables passed straight to the config file, so the user doesn't need to have a permanent config file mounted.

**Comment by yurividal** [comment]
> 
> 
> > @yurividal it would be nice if you made a pr for it? Or send me a changelist
> > Will do. I already have the cron part working. It's on my github.
> > Now I'm trying to figure out how to get the runtime environment variables passed straight to the config file, so the user doesn't need to have a permanent config file mounted.

The image is posted [here](https://hub.docker.com/r/yurividal/osticket-apache)

I had to do quite a lot of mod to your code, and used some of "campbellsoftware" install script to get it to work.

Souce code [here](https://github.com/yurividal/osticket-apache)
